### ISSUE TITLE
Could not launch terminal for debugger

### ISSUE THREAD

**Issue by cristiano-di-buduo** [eranif/codelite#1876]
I just moved from Ubuntu to Debian (stretch) and reinstalled CodeLite from apt repository as the docs say to and as I did in Ubuntu before. The program seems fully functional, except the 'Could not launch terminal for debugger' message displaying in stead of what's expected when I hit F5.

**Comment by cristiano-di-buduo** [comment]
More: the terminal setting which currently is 'konsole' cannot be selected/modified in the Settings window. The checkbox 'Use this terminal...' is selected, but displayed half-gray and cannot be manipulated.

**Comment by eranif** [comment]
CodeLite uses the `x-terminal-emulator` entry in `update-alternatives` you should configure it to use your selected terminal (in your case: `konsole`) and CodeLite will launch it

**Comment by hunar1997** [comment]
Hi, I'm on [Ubuntu 17.10 64bit] and this also happens to me, i can't debug .. codelite 10.0.0 works, i just built the latest github code (codeblocks 11.0.8) and it shows that error when trying to debug, i tried the Weekly Build  (codeblocks 11.0.7) and there when i try to debug, the program crashed with a blank terminal open .. how to do that (update-alternative) i'm not very experienced in linux :)

**Comment by dghart** [comment]
First install a terminal that is known to work e.g.  xterm, konsole, lxterminal. Then open a terminal and do:
sudo update-alternatives --config x-terminal-emulator
and select  that terminal from the displayed list.

**Comment by hunar1997** [comment]
That didn't do anything :/ , running programs is fine and terminal shows the output but i only have problem with debugging

**Comment by dghart** [comment]
>That didn't do anything

What didn't, exactly?
Which terminal did you select there? And did it work to select the one of your choice? You can find out by:
1) Opening a terminal in the usual way, then type  x-terminal-emulator <Enter> and see which terminal appears.
2) Again open a terminal. Type:
ls -l /etc/alternatives/x-terminal-emulator
I get the output:
 lrwxrwxrwx 1 root root 19 Nov 11 15:59 /etc/alternatives/x-terminal-emulator -> /usr/bin/lxterminal
and indeed my preferred terminal for debugging in CodeLite is lxterminal.

What results do you get?

**Comment by hunar1997** [comment]
It was gnome-terminal before, i changed it to xterm and nothing changed, now i installed and selected lxterminal and nothing changed again, same thing is happening:   
when i (run) it works, when i (debug) it hangs for afew seconds and shows this, look  
![screenshot from 2018-01-23 19-31-54](https://user-images.githubusercontent.com/14861276/35287829-56611444-0074-11e8-9379-f702667f13e2.png)

**Comment by dghart** [comment]
>why is it working on (codelite 10.0.0)

A very good question. The script that you are actually calling, /usr/bin/codelite_xterm, was changed a couple of years ago, but that was before the 10.0 release so you should already be using current version.

Two more suggestions:
1) Try running it direct. In a terminal do:
/usr/bin/codelite_xterm "Test" kwrite
(if you don't have kwrite installed, substitute some program that you do have.)
Does that run? If so, try it with the program you're trying to debug. If not, there's something wrong with codelite_xterm or its permissions.

2) Maybe some configuration file in your CodeLite dir is faulty. With CodeLite closed, try renaming the dir temporarily to (e.g.) /home/humar/.codeliteOld/  Now run CodeLite again (you'll need to do some configuration of course) and then try debugging your program.

**Comment by hunar1997** [comment]
I tried `/usr/bin/codelite_xterm "Test" kwrite` it works without any problems,   
> try it with the program you're trying to debug    
  
i'm not sure if i understand that correctly but i changed codelite to use that terminal and now (run) doesn't show anything (i made an infinite loop and still nothing so it doesn't open at all, and (debug) is still showing that message
   
I launched codelite with sudo nothing was different.
   
i renamed that folder and restarted and did the configurations and it's the same as  before   
i installed ubuntu 16.04 in virtual box and compiled latest code, in there (run) works but (debug) crashes the program instead of showing that message :(, but why (run) works, they don't use the same terminal?   
can you make a fresh ubuntu in a virtual machine and test it to see if you get the same thing :( i'm about to giveup

**Comment by dghart** [comment]
>I tried /usr/bin/codelite_xterm "Test" kwrite it works without any problems

Good! That means there isn't a problem with codelite_xterm itself, or the terminal that it calls.

>> try it with the program you're trying to debug
>i'm not sure if i understand that correctl

I meant, suppose the full path to your program binary is /home/hunar/CodeLite/junk/main/helloworld. Then you would do:
/usr/bin/codelite_xterm "Test"  /home/hunar/CodeLite/junk/main/helloworld

If that 'works' it shows that the problem isn't in helloword.
Or at least it might give a useful error message.

> but why (run) works, they don't use the same terminal

No they don't necessarily. And they call their terminals in different ways.

Which reminds me of a different bug-report, which turned out to be due to the full path to the binary not being passed to the terminal. Can you paste your Project settings  General page, or a screenshot of it?

**Comment by hunar1997** [comment]
I made a [short screen-cast](https://youtu.be/1SevN_JEWM8) i think its better than text :| if that didn't help then i suggest you enter my computer with [teamviewer](http://teamviewer.com) software at some point because .. i don't know what else to do

**Comment by dghart** [comment]
Your screencast didn't seem to add any useful information :(

Please upload a screenshot of your Project Settings General tab. Here's one of a template project so you can see what I mean: http://img.vim-cn.com/30/898e8b5af1de6cd7a6222d6012d3f291f34ede.png

**Comment by dghart** [comment]
The one I sent was from a template project; I wasn't suggesting it's what you should use ;)

Instead make sure that the 'Executable to Run/Debug' field is a full path (as yours was) to the binary. That can be done either by writing it in, or  by setting the Intermediate folder field to that (full) dir and then putting $(IntermediateDirectory)/$(ProjectName) into the  'Executable to Run/Debug' field (and also the 'Output File' field; yours was empty).

> I meant, suppose the full path to your program binary is /home/hunar/CodeLite/junk/main/helloworld. Then you would do:
> /usr/bin/codelite_xterm "Test" /home/hunar/CodeLite/junk/main/helloworld 
> If that 'works' it shows that the problem isn't in helloword.
> Or at least it might give a useful error message.

Did you try that?

**Comment by hunar1997** [comment]
I just tried that, (run) says `Program exited with return code: -117` and (debug) is as usual that error message

**Comment by dghart** [comment]
Hmm, so you find that this works OK:
` /usr/bin/codelite_xterm "Test" kwrite`
but this fails (when the correct filepath is used):
`/usr/bin/codelite_xterm "Test" /home/hunar/CodeLite/junk/main/helloworld`
Do I understand you correctly?

**Comment by hunar1997** [comment]
Yes exactly, and now i tried opening codelite_xterm seperately and i ran `./helloworld`in it and it's fine, wait wait :| so running `codelite_xterm "Test" helloworld` opens a blank terminal, and running `codelite_xterm "Test" ./helloworld"` does nothing, it actually closes immediately without showing the output in codelite_xterm, so it does work when i (run) it. what i just said was outside codelite, inside codelite when i enable the inbuilt terminal the (run) is fine but not compile, what code is used to debug i want to try that outside codelite

**Comment by dghart** [comment]
/usr/bin/codelite_xterm is a script, so you can alter it yourself to view the command that's passed. But first:

I just built the latest CodeLite from git in a ubuntu 17.10 virtualbox guest. Debugging works fine with gdb (it fails with lldb, but that's not unusual).

I suggest you build CodeLite as I did:
In a clean dir do:
`(mkdir build-dbg && cd build-dbg && cmake -G "Unix Makefiles" -DCMAKE_BUILD_TYPE=DebugFull -DWITH_PCH=0 -DCL_PREFIX=$(pwd)/../CL-install .. && make -j8 && make install)`
(change the -j8 to the correct figure for your machine)
What that does is to make a debug build and install it in a subdir of the source dir. You can then run it by doing:
CL-install/bin/codelite 
Create a new workspace and a  'wxWidgets Gui Application' project. Build it, then see if debugging now works.

**Comment by hunar1997** [comment]
I THINK I FOUND SOMETHING :D, i tried yours 'wxWidgets Gui Application' and everything was fine, i tried what i was trying before 'Simple Executable (g++)' with cmake as build system and it crashed, i tried it again but with (Default) and it worked, then i tried with (GNU makefile onestep build) and it doesn't even build as i reported here #1878, so the reason that it wasn't working before is that i was chosing cmake as build system, but now that i have tried i see that only (Default) works
